src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/ReindexJobTask.cs

```diff
@@ -240,6 +258,8 @@ private async Task<bool> TryPopulateNewJobFields(CancellationToken cancellationT
                     OperationOutcomeConstants.IssueType.Informational,
                     Core.Resources.NoSearchParametersNeededToBeIndexed));
                 _reindexJobRecord.CanceledTime = Clock.UtcNow;
+
+                await UpdateParametersAndCompleteJob(cancellationToken);
```

Comment:
Why are we adding this line?  It comes here only if there are NO search parameters and no resources to index.
Example: 
If there is a SP 'ABC' with status = Supported and it doesn't have any resources to reindex.
notYetIndexedParams will have 'ABC' in it.  this condition will not be true if (!notYetIndexedParams.Any() && resourceList.Count == 0).

Reply:
Can you please confirm if you tested below scenario with your code changes?

- Pick a valid resource type, that doesn't have resources in your DB yet.
- Create a custom SP and refer to the above resource type as Base Resource.
- Run a full reindex.
- Test the status of the parameter.
- Repeat the same steps for a new SP but reindex targeted SP and check the status of the parameter.

Reply:
Thanks. I removed that one.  I did re test those scenarios and they look good.